deepspeed/inference/engine.py

```diff
@@ -583,6 +605,14 @@ def _local_cuda_graph_used(self, module):
 
             return sub_module_cuda_graph
 
+    def _check_for_workspace_invalidation(self, input_signature):
+        batch_size = input_signature[1][0][0]
```

Comment:
Not confident that this will always be the correct way to retrieve batch size. Is there a better way?